lib/auth/assist/assistv1/test/service_test.go

```diff
@@ -358,36 +424,96 @@ func initSvc(t *testing.T) (map[string]context.Context, *assistv1.Service) {
 		ctxs[user.GetName()] = ctx
 	}
 
+	embedder := ai.MockEmbedder{
+		TimesCalled: make(map[string]int),
+	}
+
+	embeddings := &ai.SimpleRetriever{}
+	embeddingSrv := local.NewEmbeddingsService(backend)
 	svc, err := assistv1.NewService(&assistv1.ServiceConfig{
-		Backend:        local.NewAssistService(backend),
-		Authorizer:     authorizer,
-		Embeddings:     &ai.SimpleRetriever{},
-		ResourceGetter: &resourceGetterFake{},
+		Backend:    local.NewAssistService(backend),
+		Authorizer: authorizer,
+		Embeddings: embeddings,
+		ResourceGetter: &resourceGetterAllImpl{
+			PresenceService: presenceSvc,
+			AccessLists:     accesslistSvc,
+		},
+		Embedder: &embedder,
 	})
 	require.NoError(t, err)
 
+	unifiedResourcesCache, err := services.NewUnifiedResourceCache(context.Background(), services.UnifiedResourceCacheConfig{
+		ResourceWatcherConfig: services.ResourceWatcherConfig{
+			QueueSize:    defaults.UnifiedResourcesQueueSize,
+			Component:    teleport.ComponentUnifiedResource,
+			Client:       eventService,
+			MaxStaleness: time.Second,
+		},
+		ResourceGetter: &resourceGetterAllImpl{
+			PresenceService: presenceSvc,
+			AccessLists:     accesslistSvc,
+		},
+	})
+	require.NoError(t, err)
+
+	log.Debugf("Starting embedding watcher")
+	embeddingProcessor := ai.NewEmbeddingProcessor(&ai.EmbeddingProcessorConfig{
+		AIClient:            &embedder,
+		EmbeddingsRetriever: embeddings,
+		EmbeddingSrv:        embeddingSrv,
+		NodeSrv:             unifiedResourcesCache,
+		Jitter:              retryutils.NewFullJitter(),
+		Log:                 log.NewEntry(log.StandardLogger()),
+	})
+	log.Debugf("Starting embedding processor")
+	go embeddingProcessor.Run(context.Background(), time.Second, time.Second*10)
```

Comment:
If this context is never canceled, this go routine will never terminate.

Reply:
fixed